frame/nft-fractionalization/src/lib.rs

```diff
@@ -0,0 +1,370 @@
+pub use pallet::*;
+pub use scale_info::Type;
+pub use types::*;
+pub use weights::WeightInfo;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_system::pallet_prelude::*;
+	use scale_info::prelude::{format, string::String};
+	use sp_runtime::traits::{One, Zero};
+	use sp_std::{fmt::Display, prelude::*};
+
+	use frame_support::{
+		dispatch::DispatchResult,
+		pallet_prelude::*,
+		sp_runtime::traits::{AccountIdConversion, StaticLookup},
+		traits::{
+			fungibles::{
+				metadata::{CalcMetadataDeposit, Mutate as MutateMetadata},
+				Create, Destroy, Inspect, Mutate,
+			},
+			tokens::{
+				nonfungibles_v2::{Inspect as NonFungiblesInspect, Transfer},
+				AssetId, Balance as AssetBalance,
+			},
+			Currency, ExistenceRequirement, ReservableCurrency,
+		},
+		PalletId,
+	};
+
+	pub type AssetIdOf<T> =
+		<<T as Config>::Assets as Inspect<<T as SystemConfig>::AccountId>>::AssetId;
+	pub type AssetBalanceOf<T> =
+		<<T as Config>::Assets as Inspect<<T as SystemConfig>::AccountId>>::Balance;
+	pub type DepositOf<T> =
+		<<T as Config>::Currency as Currency<<T as SystemConfig>::AccountId>>::Balance;
+	pub type AccountIdLookupOf<T> = <<T as SystemConfig>::Lookup as StaticLookup>::Source;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// The overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The currency mechanism, used for paying for deposits.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		#[pallet::constant]
+		type Deposit: Get<DepositOf<Self>>;
+
+		/// Identifier for the collection of NFT.
+		type NftCollectionId: Member + Parameter + MaxEncodedLen + Copy + Display;
+
+		/// The type used to identify an NFT within a collection.
+		type NftId: Member + Parameter + MaxEncodedLen + Copy + Display;
+
+		/// The type used to describe the amount of fractions converted into assets.
+		type AssetBalance: AssetBalance;
+
+		/// The type used to identify the assets created during fractionalization.
+		type AssetId: AssetId;
+
+		/// Registry for the minted assets.
+		type Assets: Inspect<Self::AccountId, AssetId = Self::AssetId, Balance = Self::AssetBalance>
+			+ Create<Self::AccountId>
+			+ Destroy<Self::AccountId>
+			+ Mutate<Self::AccountId>
+			+ MutateMetadata<Self::AccountId>
+			+ CalcMetadataDeposit<<Self::Currency as Currency<Self::AccountId>>::Balance>;
+
+		/// Registry for minted NFTs.
+		type Nfts: NonFungiblesInspect<
+				Self::AccountId,
+				ItemId = Self::NftId,
+				CollectionId = Self::NftCollectionId,
+			> + Transfer<Self::AccountId>;
+
+		/// The pallet's id, used for deriving its sovereign account ID.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// The newly created asset's symbol.
+		#[pallet::constant]
+		type NewAssetSymbol: Get<Vec<u8>>;
```

Comment:
Is there any reason why these could not be bounded vecs?

Reply:
Yeah, I think this should be bound by whatever metadata/string limits are representing the assets.